### PR TITLE
New Realm Service

### PR DESCRIPTION
**Pull request by Scuilion**
Added read, update, and delete Realms.

### CODE REVIEW

**Comment by Azuka** [comment]
@Scuilion, I'm really sorry I did not get the notification for this. I'm reviewing now.

File: Readme.md [hunk 1]
```diff
@@ -37,6 +37,12 @@ make integration
 - [ ] Key
 - [ ] Protocol Mappers
 - [ ] Realms Admin
+  - [x] Get realm
+  - [~] Import realm
```
Comment (Azuka) [inline Readme.md]: Doesn't look like GH markdown supports partial checkboxes.

File: keycloak/realm_service.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+// Create realm with realm, known in Keycloak as import
+func (rs *RealmService) Create(ctx context.Context, realm *RealmRepresentation) error {
+	path := "/realms"
+	response, err := rs.client.newRequest(ctx).
+		SetBody(realm).
+		Post(path)
+
+	if err != nil {
+		fmt.Printf("\nResponse: %v", response.String())
```
Comment (Azuka) [inline keycloak/realm_service.go]: Can we get remove this? There's a Debug method on the client that shows verbose logs.

File: keycloak/realm_service.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		return err
+	}
+
+	fmt.Printf("\nResponse: %v", response.Body())
```
Comment (Azuka) [inline keycloak/realm_service.go]: As above.

File: keycloak/realms.go [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+	Protocol        string                         `json:"protocol,omitempty"`
+	ProtocolMappers []ProtocolMapperRepresentation `json:"protocolMappers,omitempty"`
+}
+
+// IdentityProviderMapperRepresentation Identity Provider Mapper
+type IdentityProviderMapperRepresentation struct {
+	Config                 AttributeMap `json:"config,ommitempty,omitempty"`
+	ID                     string       `json:"id,ommitempty,omitempty"`
+	IdentityProviderAlias  string       `json:"identityProviderAlias,ommitempty,omitempty"`
+	IdentityProviderMapper string       `json:"identityProviderMapper,ommitempty,omitempty"`
```
Comment (Azuka) [inline keycloak/realms.go]: Could we remove all instances of `,ommitempty`? Looks like a typo on original go generate.

File: keycloak/utils.go [hunk 5]
```diff
@@ -0,0 +1,11 @@
+//go:generate gomodifytags -file $GOFILE -struct MultivaluedHashMap -add-options json=omitempty -add-tags json -w -transform camelcase
+//go:generate easyjson -all $GOFILE
+
+package keycloak
+
+// MultivaluedHashMap multivalued map
+type MultivaluedHashMap struct {
```
Comment (Azuka) [inline keycloak/utils.go]: This can probably go in `types.go`

File: keycloak/realms.go (hunk 4 above)
Comment (Scuilion) [reply keycloak/realms.go]: You want me to remove `,omitempty` for all structs or just the ones I added?

File: keycloak/utils.go (hunk 5 above)
Comment (Scuilion) [reply keycloak/utils.go]: :+1:

File: Readme.md (hunk 1 above)
Comment (Scuilion) [reply Readme.md]: Since import realm is not fully featured, I'll leave it empty.

File: keycloak/realms.go (hunk 4 above)
Comment (Scuilion) [reply keycloak/realms.go]: oh, missed that. Thanks!

**Comment by Scuilion** [comment]
I believe the linting is failing because of this recent change to go.uuid
https://github.com/satori/go.uuid/issues/18

**Comment by Azuka** [comment]
@Scuilion, I'm sorry for the delay. Can you pull in the latest changes? After that I think we're good to go.

**Comment by Scuilion** [comment]
Not sure why the dependency can't be resolved.

**Comment by Azuka** [comment]
Actually, I went ahead and fixed. Thanks for the work!

**Pull request merged**